shell/machine-config/pnap.vue

```diff
@@ -0,0 +1,279 @@
+import { LabeledInput } from '@components/Form/LabeledInput';
+
+export default {
+  components: {
+    Loading,
+    Banner,
+    LabeledSelect,
+    LabeledInput,
+  },
+
+  mixins: [CreateEditView],
+
+  props: {
+    credentialId: {
+      type:      String,
+      required:  true,
+    },
+    disabled: {
+      type:     Boolean,
+      default:  false,
+    },
+  },
+
+  async fetch() {
+    this.errors = [];
+
+    try {
+      if (this.credentialId) {
+        this.credential = await this.$store.dispatch('rancher/find', {
+          type: NORMAN.CLOUD_CREDENTIAL,
+          id:   this.credentialId,
+        });
+        this.$set(this.value, 'clientIdentifier', this.credential.decodedData.clientIdentifier);
+      }
+    } catch (e) {
+      this.credential = null;
+    }
+
+    try {
+      this.allProducts = await this.$store.dispatch('pnap/allProducts', { credentialId: this.credentialId });
+
+      this.osChoices = await this.$store.dispatch('pnap/osChoices', { credentialId: this.credentialId });
+
+      const serverOs = this.osChoices[0].value;
+
+      if (!this.value.serverOs) {
+        this.value.serverOs = serverOs;
+      }
+      if (!this.value.serverLocation) {
+        this.value.serverLocation = 'PHX';
+      }
+      if (!this.value.serverType) {
+        this.value.serverType = 's1.c1.medium';
+      }
+      this.updateInterfaces();
+    } catch (e) {
+      this.errors = exceptionToErrorsArray(e);
+    }
+  },
+
+  data() {
+    return {
+      credential:      null,
+      allProducts:     null,
+      osChoices:       null,
+    };
+  },
+
+  watch: {
+    'credentialId'() {
+      this.$fetch();
+    },
+  },
+
+  computed: {
+    isCreate() {
+      return this.mode === _CREATE;
+    },
+    locationChoices() {
+      const products = this.allProducts;
+      const result = [];
+
+      if (Array.isArray(products)) {
+        const map = new Map();
+
+        for (const prod of products) {
+          for (const plan of prod.plans) {
+            if (!map.has(plan.location)) {
+              map.set(plan.location, true);
+              result.push({ value: plan.location, label: plan.location });
+            }
+          }
+        }
+      } else {
+        result.push({ value: products.message });
+      }
+
+      return result;
+    },
+    typeChoices() {
+      const products = this.allProducts;
+      const result = [];
+
+      if (Array.isArray(products)) {
+        const map = new Map();
+        const serverLocation = this.value.serverLocation;
+
+        for (const prod of products) {
+          for (const plan of prod.plans) {
+            if (
+              !map.has(prod.productCode) &&
+              plan.location === serverLocation
+            ) {
+              map.set(prod.productCode, true);
+              result.push({
+                value: prod.productCode,
+                label: `${ prod.productCode } - $${ plan.price }/hour`,
```

Comment:
No change required, just confirmation, pnap only charge in USD correct?

Reply:
Pnap can charge EU customers in euros, but all prices are in dollars and just converted.